### PR TITLE
initial commit of manager home page

### PR DESCRIPTION
**Pull request by athulmuraleedharan**

### CODE REVIEW

File: client/idp/src/app/manager/homepage/homepage.component.css [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    left: 05 em;
+}
+
+h2 {
+    font-size: 36px;
+    text-align: center;
+    margin-top: 20px;
+}
+
+h6 {
```
Comment (mishelshaji) [inline client/idp/src/app/manager/homepage/homepage.component.css]: Why not use the bootstrap `text-center` class instead?

File: client/idp/src/app/manager/homepage/homepage.component.css [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+.container-fluid {
+    background-color: rgb(246, 247, 249);
+    padding: 2rem;
+
+}
+
+.text-left {
+    font-weight: bolder;
+    color: black;
```
Comment (mishelshaji) [inline client/idp/src/app/manager/homepage/homepage.component.css]: Use color value instead.

File: client/idp/src/app/manager/homepage/homepage.component.css [hunk 3]
```diff
@@ -0,0 +1,70 @@
+.bg-opacity {
+    opacity: 80%;
+    color: black
+}
+
+.container-fluid {
+    background-color: rgb(246, 247, 249);
+    padding: 2rem;
+
```
Comment (mishelshaji) [inline client/idp/src/app/manager/homepage/homepage.component.css]: Remove the blank line.

File: client/idp/src/app/manager/homepage/homepage.component.css [hunk 4]
```diff
@@ -0,0 +1,70 @@
+.bg-opacity {
+    opacity: 80%;
+    color: black
```
Comment (mishelshaji) [inline client/idp/src/app/manager/homepage/homepage.component.css]: Give color values.

File: client/idp/src/app/manager/homepage/homepage.component.css [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    background-color: rgb(239, 243, 254);
+}
+
+.width {
+    width: 20rem;
+}
+
+.inbox-messages {
+    width: 30rem;
+
```
Comment (mishelshaji) [inline client/idp/src/app/manager/homepage/homepage.component.css]: Remove the blank lines.

File: client/idp/src/app/manager/homepage/homepage.component.ts [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+  styleUrls: ['./homepage.component.css'],
+})
+export class HomepageComponent implements OnInit {
+  counters = [
+    { name: 'Total Empolyees', count: 1000, limit: 1500 },
+    { name: 'Total Created IDP', count: 0, limit: 126 },
+    { name: 'Total Pending IDP', count: 0, limit: 600 },
+    { name: 'Total Completed IDP', count: 0, limit: 660 },
+  ];
+  IntervalIds: any[] = [];
```
Comment (mishelshaji) [inline client/idp/src/app/manager/homepage/homepage.component.ts]: Use proper naming conventions. Also, add a blank line above.

File: client/idp/src/app/manager/homepage/homepage.component.ts [hunk 7]
```diff
@@ -0,0 +1,152 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-homepage',
+  templateUrl: './homepage.component.html',
+  styleUrls: ['./homepage.component.css'],
+})
+export class HomepageComponent implements OnInit {
+  counters = [
```
Comment (mishelshaji) [inline client/idp/src/app/manager/homepage/homepage.component.ts:11]: Add a blank line above.

File: client/idp/src/app/manager/homepage/homepage.component.ts [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+      subject: 'Urgent Message',
+      text: 'Praesent vestibulum dictum dui, eu gravida justo placerat a.',
+    },
+    {
+      sender: 'Bob Smith',
+      subject: 'Follow-Up',
+      text: 'Vivamus eget velit vel libero mollis consectetur.',
+    },
+  ];
+  users = [
```
Comment (mishelshaji) [inline client/idp/src/app/manager/homepage/homepage.component.ts:54]: Add a blank line above.

File: client/idp/src/app/manager/homepage/homepage.component.ts [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    {
+      name: 'Jim Smith',
+      profile: 'https://picsum.photos/100',
+      department: 'IT',
+      objective: 'Improve system performance by 20%',
+      idp: 'IMP-ID-200908-2022',
+      remark: 'pending',
+    },
+  ];
+  constructor() {}
```
Comment (mishelshaji) [inline client/idp/src/app/manager/homepage/homepage.component.ts:153]: Add a blank line above.

File: client/idp/src/app/manager/homepage/homepage.component.ts [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+  lastMessage(message: any) {
```
Comment (mishelshaji) [inline client/idp/src/app/manager/homepage/homepage.component.ts:155]: Add a blank line above.

**Review by mishelshaji — CHANGES_REQUESTED** [review]

File: client/idp/src/app/manager/homepage/homepage.component.css (hunk 1 above)
Comment (athulmuraleedharan) [reply client/idp/src/app/manager/homepage/homepage.component.css]: It has changed into bootstrap.

File: client/idp/src/app/manager/homepage/homepage.component.css (hunk 2 above)
Comment (athulmuraleedharan) [reply client/idp/src/app/manager/homepage/homepage.component.css]: color had changed into  color value.

File: client/idp/src/app/manager/homepage/homepage.component.css (hunk 3 above)
Comment (athulmuraleedharan) [reply client/idp/src/app/manager/homepage/homepage.component.css]: Blank line had removed.

File: client/idp/src/app/manager/homepage/homepage.component.css (hunk 4 above)
Comment (athulmuraleedharan) [reply client/idp/src/app/manager/homepage/homepage.component.css]: Changed into values.

File: client/idp/src/app/manager/homepage/homepage.component.css (hunk 5 above)
Comment (athulmuraleedharan) [reply client/idp/src/app/manager/homepage/homepage.component.css]: Blank lines had removed.

File: client/idp/src/app/manager/homepage/homepage.component.ts (hunk 6 above)
Comment (athulmuraleedharan) [reply client/idp/src/app/manager/homepage/homepage.component.ts]: Issue had cleared.

File: client/idp/src/app/manager/homepage/homepage.component.ts (hunk 7 above)
Comment (athulmuraleedharan) [reply client/idp/src/app/manager/homepage/homepage.component.ts:11]: Blank line added.

File: client/idp/src/app/manager/manager-routing.module.ts [hunk 11]
```diff
@@ -1,18 +1,20 @@
...
 const routes: Routes = [
   {
-    path: '', component: ManagerLayoutComponent, children: [
-      { path: 'home', component:  HomeComponent},
-    ]
-  }
+    path: '',
+    component: ManagerLayoutComponent,
+    children: [
+      { path: 'manager-homepage', component: HomepageComponent },
```
Comment (mishelshaji) [inline client/idp/src/app/manager/manager-routing.module.ts]: Remove the trailing comma.

File: client/idp/src/app/manager/manager-routing.module.ts [hunk 12]
```diff
@@ -1,18 +1,20 @@
...
+    ],
+  },
 ];
 
 @NgModule({
   imports: [RouterModule.forChild(routes)],
-  exports: [RouterModule]
+  exports: [RouterModule],
```
Comment (mishelshaji) [inline client/idp/src/app/manager/manager-routing.module.ts]: Remove the trailing comma.

**Review by mishelshaji — CHANGES_REQUESTED** [review]

File: client/idp/src/app/manager/manager-routing.module.ts (hunk 11 above)
Comment (athulmuraleedharan) [reply client/idp/src/app/manager/manager-routing.module.ts]: It had Removed.

File: client/idp/src/app/manager/manager-routing.module.ts (hunk 12 above)
Comment (athulmuraleedharan) [reply client/idp/src/app/manager/manager-routing.module.ts]: Comma had removed.

**Review by maheshp2002 — APPROVED** [review]

File: client/idp/src/app/manager/manager-routing.module.ts [hunk 13]
```diff
@@ -1,18 +1,20 @@
...
 const routes: Routes = [
   {
-    path: '', component: ManagerLayoutComponent, children: [
-      { path: 'home', component:  HomeComponent},
-    ]
-  }
+    path: '',
+    component: ManagerLayoutComponent,
+    children: [
+      { path: 'manager-homepage', component: HomepageComponent }
```
Comment (mishelshaji) [inline client/idp/src/app/manager/manager-routing.module.ts]: Why not use simply `manager` or `manager/home` as the URL?

**Review by mishelshaji — CHANGES_REQUESTED** [review]

File: client/idp/src/app/app-routing.module.ts [hunk 14]
```diff
@@ -14,6 +15,9 @@ const routes: Routes = [
     path: 'admin', 
     loadChildren: () => import('./admin/admin.module').then(m => m.AdminModule)
   },
+  {
+    path: 'invalid', component: InvalidpageComponent,
```
Comment (mishelshaji) [inline client/idp/src/app/app-routing.module.ts]: What is the purpose of this page?

File: client/idp/src/app/shared/invalidpage/invalidpage.component.ts [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+
+@Component({
+  selector: 'app-invalidpage',
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './invalidpage.component.html',
+  styleUrls: ['./invalidpage.component.css']
+})
+export class InvalidpageComponent {
+
```
Comment (mishelshaji) [inline client/idp/src/app/shared/invalidpage/invalidpage.component.ts]: Remove the extra blank lines.

**Review by mishelshaji — CHANGES_REQUESTED** [review]

File: client/idp/src/app/app-routing.module.ts (hunk 14 above)
Comment (athulmuraleedharan) [reply client/idp/src/app/app-routing.module.ts]: This is the Not found page.

**Review by mishelshaji — APPROVED** [review]

**Pull request merged**